### ISSUE TITLE
Collecting KroneckerSums is slower than using kron

### ISSUE THREAD

**Issue by elisno** [MichielStock/Kronecker.jl#26]
The following snippet compares `collect` and `kron` for evaluating kronecker sums (of order 2 and 3).
```julia
using Kronecker, Test, BenchmarkTools

N = 10
A,B,C = (rand(N,N) for _ in 1:3)

# Define collect functions
function foo(A,B)
    collect(A ⊕ B)
end

function bar(A,B)
    IA,IB = oneunit.((A,B))
    kron(A,IB) + kron(IA,B)
end

function foo3(A,B,C)
    collect(A ⊕ B ⊕ C)
end

function bar3(A,B,C)
    IA, IB, IC = oneunit.((A,B,C))
    kron(A,IB,IC) + kron(IA,B,IC) + kron(IA,IB,C)
end


println("Timing A ⊕ B")
println("⊕")
@btime foo($A,$B)
println("kron")
@btime bar($A,$B)

println("\n Timing A ⊕ B ⊕ C")
println("⊕")
@btime foo3($A,$B,$C)
println("kron")
@btime bar3($A,$B,$C)


# Compare collections
@test foo(A,B) == bar(A,B)
@test foo3(A,B,C) == bar3(A,B,C)
```

Running this gives the output:

```julia
Timing A ⊕ B
⊕
  217.697 μs (3 allocations: 78.23 KiB)
kron
  58.415 μs (8 allocations: 236.36 KiB)

 Timing A ⊕ B ⊕ C
⊕
  30.457 ms (4 allocations: 7.63 MiB)
kron
  11.000 ms (18 allocations: 30.75 MiB)
```

**Comment by MichielStock** [comment]
Resolved in 3ef77155b902d0ce188a595154bf21f0055031c4, `⊕` is now slightly faster than `kron`.

```julia
julia> @btime foo3($A,$B,$C);
  4.817 ms (22 allocations: 23.20 MiB)

julia> @btime bar3($A,$B,$C);
  6.008 ms (18 allocations: 30.75 MiB)
```

**Comment by elisno** [comment]
I ran the snippet again on https://github.com/MichielStock/Kronecker.jl/commit/3ef77155b902d0ce188a595154bf21f0055031c4 and got the following timings:

```julia
Timing A ⊕ B
⊕
  57.926 μs (11 allocations: 238.14 KiB)
kron
  59.464 μs (8 allocations: 236.36 KiB)

 Timing A ⊕ B ⊕ C
⊕
  7.345 ms (22 allocations: 23.20 MiB)
kron
  11.044 ms (18 allocations: 30.75 MiB)

```
`collect` gets ~4x speedup! Great work!
